consul/config.sls

```diff
@@ -1,18 +1,24 @@
 {%- from slspath + '/map.jinja' import consul with context -%}
 
 consul-config:
-  file.serialize:
```

Comment:
IMHO, manage structured files with `file.managed` is really bad (I tried this myself and it always requires many hacks and checks).

I'd suggest to keep using `file.serialize`, and create another check for syntax specifically.
We could use [mod_run_check_cmd](https://docs.saltstack.com/en/2017.7/ref/states/all/salt.states.file.html#salt.states.file.mod_run_check_cmd), but just realized it doesn't work as expected (it's used as an internal function for `file.managed`. It's already exposed externally, however it doesn't met functions requirements, where it doesn't have `name,changes` in returned output).

So till fixing that (mostly I will make a PR for that), let's use `cmd.run` with `unless`:
```
consul-config-validate: 
  cmd.run:
    - name: /usr/local/bin/consul validate /etc/consul.d/config.json
    - unless: /usr/local/bin/consul validate /etc/consul.d/config.json
```

The `unless` clue, will prevent running the command when it has a valid syntax. So it avoids any extra noise.